### PR TITLE
PLAT-135587: Updated Silicon skin styles for Button

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Button component needs to be updated to the latest design changes

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Added styling for Button based on the latest UX design for Silicon
- Because of the new styling for iconOnly Button, other components were affected, so new styles were added to restore those styles to the current value from the develop branch ( ColorPicker, ColorPicker/SwatchButton, Keypad, IncrementSlider/IncrementSliderButton )
- Because @agate-button-height has changed for silicon, its' usage DateTimePicker, SliderButton and internal/Picker was changed to a static value(the previous value of agate-button-height)
- @silicon-gradient core color has a new value on the UX guide. The change of its' value got many ss-tests to fail, but this is normal
- the AgateSilicon ss-tests for other components that use Button or Icon are failing because those components use Button or Icon. They will be fixed when those component`s design will be updated.
- update .active() mixin to include also pressed state ( active state of Button was not triggered on select Enter key until now)
- added new screenshot tests for Button

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-135587

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

**Review by stanca-pop-lgp — CHANGES_REQUESTED** [review]
Please check out the styling for selected. It does not seem to apply style from Zeplin in Storybook when "selected" knob is checked.

File: styles/variables-silicon.less [hunk 1]
```diff
@@ -37,17 +37,39 @@
...
 @agate-button-line-height: 1.17em;
+@agate-button-max-width: 420px;
+@agate-button-min-width: 180px;
+@agate-button-icon-margin-end: 9px;
+@agate-button-icon-margin-start: 0;
 @agate-button-huge-border-width: @agate-button-border-width;
+@agate-button-huge-font-size: 33px;
+@agate-button-huge-h-padding: @agate-button-h-padding;
+@agate-button-huge-height: 180px;
+@agate-button-huge-max-width: 300px;
```
Comment (stanca-pop-lgp) [inline styles/variables-silicon.less:57]: This should be 360px.

**Comment by daniel-stoian-lgp** [comment]
> Also the icon's top and bottom margins for all button sizes.

Since the icon is vertically aligned at middle with the text, apllying those padding/margins have no  visual effect

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

File: tests/screenshot/apps/components/Button.js [hunk 2]
```diff
@@ -1,10 +1,161 @@
...
+	<Button icon="arrowharpoonright" iconFlip="vertical">click me</Button>,
+	<Button icon="arrowharpoonright" iconFlip="both">click me</Button>,
+	<Button icon="arrowharpoonright" iconFlip="auto">click me</Button>,
+
+
+	// *************************************************************
+	// Tallglyph validation
+	// locale = 'vi-VN'
+	// *************************************************************
+	...withConfig({locale: 'vi-VN'}, [
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/Button.js:67]: Could you add focused case also?

...withConfig({focus: true}, [
...
]

File: styles/colors-silicon-day.less [hunk 3]
```diff
@@ -80,21 +80,28 @@
...
-@agate-button-bg-color:       @agate-white;
+@agate-button-bg-color:       @silicon-gray;
 @agate-button-bg-image:       none;
-@agate-button-border-color:   fade(@agate-black, 50%);
+@agate-button-border-color:   transparent;
 @agate-button-shadow:         none;
 @agate-button-active-color:   @agate-button-color;
-@agate-button-active-bg-color: @silicon-light-gray;
+@agate-button-active-bg-color: @silicon-gray;
+@agate-button-disabled-bg-color: fade(@agate-button-bg-color, 50%);
```
Comment (jeonghee27) [inline styles/colors-silicon-day.less:89]: 50%? I saw 30%

File: Button/Button.module.less [hunk 4]
```diff
@@ -42,14 +42,24 @@
...
+			padding: @agate-button-client-padding;
 			color: @agate-button-color;
 
 			.icon {
 				.margin-start-end(@agate-button-icon-margin-start, @agate-button-icon-margin-end);
 			}
 		}
 
+		.disabled({
+			.bg {
```
Comment (jeonghee27) [inline Button/Button.module.less:54]: For disabled style, opacity 0.6 was applied before.
This impede consistency depending on the background color.
So, GUI person said that some fixed color will apply for disabled. Could you remove the opacity style that was previously applied only to silicon?
CC : @adrian-cocoara-lgp @stanca-pop-lgp @paul-beldean-lgp @0x64 @MikyungKim

Comment (daniel-stoian-lgp) [reply Button/Button.module.less:54]: Alright. This was actually the cause of why I used 50% opacity for colors instead of the 30% from the GUI guide.

Comment (daniel-stoian-lgp) [reply Button/Button.module.less:54]: @jeonghee27  Actually, is it ok to set @agate-disabled-opacity to 1? That will affect all components. Or maybe I misunderstood which previosuly opacity of 0.6 you meant.

Comment (jeonghee27) [reply Button/Button.module.less:54]: Oh, I didn't say it clearly. First of all, I wanted to remove it only from the silicone button.
And, it still looks opacity applied. It doesn't look like #444444.
image

**Comment by jeonghee27** [comment]
The press effect looks different from the GUI document. Could you please check again?

File: Button/Button.module.less (hunk 4 above)
Comment (daniel-stoian-lgp) [reply Button/Button.module.less:55]: I don't know what happened, but apparently, the @agate-disabled-opacity: 1 setting was not pushed. I fixed now the opacity only for Silicon Button

**Comment by daniel-stoian-lgp** [comment]
> The (focus+press) effect looks different from the GUI document. Could you please check again?

Fixed it :)

**Comment by jeonghee27** [comment]
Silicon works well, but there is regression in the press effect and disable in Gallium. please check :(

**Comment by jeonghee27** [comment]
image
Regression is in the press effect and disable in Gallium. :(

After correction, please review again for the operation. @stanca-pop-lgp

**Comment by daniel-stoian-lgp** [comment]
>
> Regression is in the press effect and disable in Gallium. :(
> 
> After correction, please review again for the operation. @stanca-pop-lgp

@jeonghee I pushed a fix.
All ss-tests passed. I had to introduce an extra .vendor-opacity() styling for LabeledIcoButton because the new disabled opacity for Button had a bigger specificity and introduced a regression for a disabled LabeledIconButton.

**Comment by stanca-pop-lgp** [comment]
It seems that the text is not vertically aligned in the middle for the smallest size.

**Comment by stanca-pop-lgp** [comment]
Please also adjust styling for joinedPosition when button is smallest.

**Comment by daniel-stoian-lgp** [comment]
> It seems that the text is not vertically aligned in the middle for the smallest size. Could you check it out?

For me it looks ok

image

**Comment by stanca-pop-lgp** [comment]
> > It seems that the text is not vertically aligned in the middle for the smallest size. Could you check it out?
> 
> For me it looks ok
> 
> image

image

**Comment by daniel-stoian-lgp** [comment]
> > > It seems that the text is not vertically aligned in the middle for the smallest size. Could you check it out?
> > 
> > 
> > For me it looks ok
> > image
> 
> image
> 
> image

I can't reproduce it on my computer

image

**Comment by stanca-pop-lgp** [comment]
> > > > It seems that the text is not vertically aligned in the middle for the smallest size. Could you check it out?
> > > 
> > > 
> > > For me it looks ok
> > > image
> > 
> > 
> > image
> > image
> 
> I can't reproduce it on my computer
> 
> image

Try applying the same value to line-height as to height.

File: styles/variables-silicon.less [hunk 5]
```diff
@@ -32,22 +32,42 @@
...
+@agate-button-huge-font-size: 33px;
+@agate-button-huge-h-padding: @agate-button-h-padding;
+@agate-button-huge-height: 180px;
+@agate-button-huge-max-width: 360px;
+@agate-button-huge-min-width: 180px;
 @agate-button-small-border-radius: (@agate-button-small-height / 2);
-@agate-button-small-font-size: 28px;
-@agate-button-smallest-height: 39px;
-@agate-button-smallest-font-size: 18px;
+@agate-button-small-font-size: 25px;
```
Comment (stanca-pop-lgp) [inline styles/variables-silicon.less]: Multiple of 3. (also design says 26)

**Review by jeonghee27 — APPROVED** [review]
LGTM

**Pull request merged**